master/internal/task/allocation_test.go

```diff
@@ -100,6 +100,18 @@ func TestAllocation(t *testing.T) {
 						Devices: []device.Device{},
 					},
 				}
+				db.On("UpdateAllocationState", a.req.AllocationID, model.AllocationStateAssigned,
```

Comment:
can we weave these between the asks happening and make sure they get called correctly?